[Docs] New color guidelines page

cchaos opened the pull request:
### Summary

Hopefully more than just an exploration, this PR updates the color guidelines page to give more explanations to each color name and a better layout for scalability.

It breaks the file down into smaller re-usable components as well so it's easier to maintain.

The color explanations are currently just quick descriptions that come to mind, but if others have more they'd add please comment or commit directly 🙏 

I'd also really love to get some section headers to populate in the sidebar, but at the moment it is tricky because of some page-wide states that can't populate down using the JSON returned method.

Please take a look at the build preview.

### Checklist

- [x] Check against **all themes** for compatibility in both light and dark modes
- [x] Checked in **mobile**
- [x] Checked in **IE11** and **Firefox**
- ~[ ] Props have proper **autodocs**~
- [x] Added **documentation** examples
- ~[ ] Added or updated **jest tests**~
- ~[ ] Checked for **breaking changes** and labeled appropriately~
- [x] Checked for **accessibility** including keyboard-only and screenreader modes
- ~[ ] A [changelog](https://github.com/elastic/eui/blob/master/CHANGELOG.md) entry exists and is marked appropriately~

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_3001/

myasonik commented:
Loved the changes! 👍 

One ask before merging (which, I hope this gets merged!) - none of the new color buttons have a focus state and there doesn't seem to be a keyboard accessible way to activate the top color buttons (<kbd>shift</kbd>+<kbd>enter</kbd>/<kbd>space</kbd> is what I would expect).

cchaos commented:
I've added a focus state, thanks for the reminder! But what do you mean by "activate"? If I hit <kbd>return</kbd>, I get the color name copied, if hit <kbd>shift</kbd>+<kbd>return</kbd> it scrolls to the correct section in the page.

myasonik commented:
🤔 Hmmm, I get the color copied even if I use <kbd>shift</kbd>+<kbd>enter</kbd>... Will have to investigate more on Monday.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_3001/

myasonik commented:
It's a firefox bug... 😕

cristina-eleonora commented:
Hey @cchaos, leaving some feedback here on the new color guidelines. 

As a person who quickly scans the page and doesn't read everything, I didn't realize that that the changes I make in the slider will apply to what's shown below. Since the slider is in a rectangle with a different background color, I thought it was its own thing.

<img width="985" alt="Screenshot 2020-03-11 at 13 41 33" src="https://user-images.githubusercontent.com/44808440/76418014-3138a580-639e-11ea-8efb-b351c8bea846.png">

____

I found this slider a bit difficult to interpret, probably because we have many labels that are under 2 colors (bar), instead of just one. 

I, personally, find the half lit circle icon confusing... Perhaps an alert icon would let users know that this doesn't meet the contrast standard. 

<img width="507" alt="Screenshot 2020-03-11 at 13 40 40" src="https://user-images.githubusercontent.com/44808440/76417867-f171be00-639d-11ea-8426-27089fbfb121.png">

The slider says `minimum color combination to show` but it allows the user to show combinations that don't meet the minimum criteria. It's not clear that if we can use all of these (unless you only rely on the icons). I was hoping for the hover to offer more guidance

<img width="991" alt="Screenshot 2020-03-11 at 13 52 03" src="https://user-images.githubusercontent.com/44808440/76418782-875a1880-639f-11ea-8914-fbd0628d11ec.png">

Perhaps the hover text can say that this shouldn't be used when the contrast is this low, instead of letting users copy the SASS config.

<img width="366" alt="Screenshot 2020-03-11 at 13 54 53" src="https://user-images.githubusercontent.com/44808440/76419079-0b140500-63a0-11ea-8b19-1efbf590901d.png">

cchaos commented:
Thanks @cristina-eleonora !

> Since the slider is in a rectangle with a different background color, I thought it was its own thing.

Yeah my idea is to make this sticky to the top of the window so it stays in place as you scroll.

> I found this slider a bit difficult to interpret, probably because we have many labels that are under 2 colors (bar), instead of just one.

I'm just going to remove the colored indicators.

> I, personally, find the half lit circle icon confusing... Perhaps an alert icon would let users know that this doesn't meet the contrast standard.

I can understand this. Though `alert` is too harsh since this contrast does meet standards for graphics and large text. I'll try to come up with a different icon.

> The slider says minimum color combination to show but it allows the user to show combinations that don't meet the minimum criteria.

I'll have to check this out, it sounds like a bug

> Perhaps the hover text can say that this shouldn't be used when the contrast is this low, instead of letting users copy the SASS config.

Good idea!

cchaos commented on src-docs/src/views/guidelines/colors.js:
```diff
@@ -1,437 +1,353 @@
+      <EuiCallOut
+        color="warning"
+        iconType="accessibility"
+        title="Amsterdam changes in contrast levels">
```
Good call. I'll update. I'll probably just lose the whole callout and put a toggle next to the slider

cchaos commented:
@snide & @cristina-eleonora , Thank you so much for your feedback! I think I've addressed all of your concerns if you wouldn't mind giving it another quick pass. 🙇‍♀

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_3001/

cristina-eleonora commented:
It looks great, @cchaos ! Just 2 minor details:

- Perhaps `Minimum color contrast combinations to show`can become `Color combinations to show`
- Add more white space between the description and `Use the slider and toggle to adjust the color combinations shown in the sections below.`. Change spacer from `s` to `xl` to make sure it gets the propper attention

<img width="942" alt="Screenshot 2020-03-12 at 11 05 45" src="https://user-images.githubusercontent.com/44808440/76510350-c052c580-6451-11ea-961b-4c0275eb26e6.png">

But this is just me nitpicking :)

cchaos commented:
Ok I think I've addressed all concerns. Though I left the slider label the same since it needs that "Minimum" portion.

I'm asking for a few more reviewers.

mdefazio commented:
I think this looks good!  Super tiny edit would be to have the tooltip say "SASS copied" instead of just "copied" (or something). First time I clicked on the color I didn't give it enough time to show the first tooltip and so I wasn't sure exactly what was copied.

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_3001/

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_3001/

andreadelrio commented:
@cchaos This is awesome! I like that you added a subtle border to colours that were too light. I think that one nice-to-have would be a way to go back to the colour list after the user has used the handy "shift + clic to scroll to section" shortcut.

cchaos commented:
Thanks!

> I think that one nice-to-have would be a way to go back to the colour list after the user has used the handy "shift + clic to scroll to section" shortcut.

I'm hoping that a follow-up will be figuring out how to add sections to the side nav. Not for each color, but for faux sections like "Core colors", "Grays" to also easily scroll/link to.

cchaos commented:
Awesome, thank you all!

Jenkins, test this

kibanamachine commented:
Preview documentation changes for this PR: https://eui.elastic.co/pr_3001/
